plugins/modules/ise_radius_integration_workflow_manager.py

```diff
@@ -991,16 +1008,22 @@ def get_want_authentication_policy_server(self, auth_policy_server):
                 position_ise_creds += 1
 
             pxgrid_enabled = auth_policy_server.get("pxgrid_enabled")
-            if pxgrid_enabled is not None:
-                auth_server.update({"pxgridEnabled": pxgrid_enabled})
+            if not pxgrid_enabled:
+                if not auth_server_exists:
+                    auth_server.update({"pxgridEnabled": True})
+                else:
+                    auth_server.update({"pxgridEnabled": auth_server_details.get("pxgridEnabled")})
             else:
-                auth_server.update({"pxgridEnabled": True})
+                auth_server.update({"pxgridEnabled": pxgrid_enabled})
 
             use_dnac_cert_for_pxgrid = auth_policy_server.get("use_dnac_cert_for_pxgrid")
```

Comment:
Same as above... 
We have redundant code of calling auth_server.update(useDnacCertForPxgrid)..Instead get value and call this API at last..